fspiop-api/documents/Tracing v1.0.md

```diff
@@ -0,0 +1,895 @@
+
+##### **Table 5 – P2P Transfer Participants**
+
+Participant | Identifier | _vendor-name_
+------|-----------|----------
+Payer FSP | BankNrOne | banknrone
+Payee FSP | MobileMoney | mobilemoney
+Mojaloop Payment Switch | Switch | moja
+
+
+##### 4.1.1. Request Party Information
+
+4.1.1.1. Request Party Information from BankNrOne to Switch
+
+BankNrOne sends a **GET /parties** request to the Switch, and includes the following generated _traceparent_ information:
+
+| variable | value | description |
+| ---|---|--- |
+| _version_ | `00` | Indicates Standard version as per [Table 3](#table-3-–-supported-version-formats) is being used. |
+| _trace-id_ | `91e502e28cd723686e9940bd3f378f85` | Unique trace identifier randomly generated. Refer to [Table 2](#table-2-–-data-model-for-tracing-values). |
+| _parent-id_ | `b0f903d000944947` | Updated to represent the current span. Refer to [Table 2](#table-2-–-data-model-for-tracing-values). |
+| _trace-flags_ | `01 `| Indicates that BankNrOne has recorded the trace. |
+
+The _tracestate_ is also include with the following information:
+| index | _list-member_ |value | description|
+| --- | --- | --- | --- |
+| 0 | `banknrone` | `b0f903d000944947` | Mapped from _parent-id_. |
+
+The following trace headers will be include in the request:
+
+**Note:** The fields highlighted below as **`value`** will indicate a newly generated or modified value depending on context. Non-highlighted fields will indicate an unchanged value.
+
+> traceparent: **`00`**-**`91e502e28cd723686e9940bd3f378f85`**-**`b0f903d000944947`**-**`01`**<br/>
+> tracestate: **`banknrone`**=**`b0f903d000944947`**<br/>
+
+<br/>
+The request sample including trace headers:
+
+```http
+GET /parties/MSISDN/123456789 HTTP/1.1
+Accept: application/vnd.interoperability.parties+json;version=1
+Content-Type: application/vnd.interoperability.parties+json;version=1.0
+Date: Tue, 15 Nov 2017 10:13:37 GMT
+FSPIOP-Source: BankNrOne
+traceparent: 00-91e502e28cd723686e9940bd3f378f85-b0f903d000944947-01
+tracestate: banknrone=b0f903d000944947
+```
+
+4.1.1.2. Request Party Information from Switch to MobileMoney
+
+Switch receives the request, and mutates the trace information before forwarding the request to MobileMoney.
+
+The _traceparent_ is mutated as follows:
+
+| variable | value | description |
+| ---|---|--- |
+| _version_ | 00 | Unchanged. |
+| _trace-id_ | 91e502e28cd723686e9940bd3f378f85 | Unchanged as the Switch continuing the trace. |
+| _parent-id_ | `b0f903d000944947` | Updated to represent the current span. Refer to [Table 2](#table-2-–-data-model-for-tracing-values). |
+| _trace-flags_ | 01 | Indicates that Switch has recorded the trace. |
+
+The _tracestate_ is mutated as follows:
+
+| index | _list-member_ |value | description|
+| --- | --- | --- | --- |
+| 1 | `moja` | `f86425c1a005e5e2` | Mapped from _parent-id_. |
```

Comment:
Are we using 0 indexing or 1 indexing? This table is not consistent with 4.1.1.1 above